src/test/javascript/spec/integration/code-editor/code-editor-container.integration.spec.ts

```diff
@@ -143,44 +141,44 @@ describe('CodeEditorContainerIntegration', () => {
-
-                const codeEditorRepositoryService = containerDebugElement.injector.get(CodeEditorRepositoryService);
-                const codeEditorRepositoryFileService = containerDebugElement.injector.get(CodeEditorRepositoryFileService);
-                const participationWebsocketService = containerDebugElement.injector.get(ParticipationWebsocketService);
-                const resultService = containerDebugElement.injector.get(ResultService);
-                const buildLogService = containerDebugElement.injector.get(CodeEditorBuildLogService);
-                const programmingExerciseParticipationService = containerDebugElement.injector.get(ProgrammingExerciseParticipationService);
-                conflictService = containerDebugElement.injector.get(CodeEditorConflictStateService);
-                domainService = containerDebugElement.injector.get(DomainService);
-                const submissionService = containerDebugElement.injector.get(ProgrammingSubmissionService);
-
-                subscribeForLatestResultOfParticipationSubject = new BehaviorSubject<Result | undefined>(undefined);
-
-                getLatestPendingSubmissionSubject = new Subject<ProgrammingSubmissionStateObj>();
-
-                checkIfRepositoryIsCleanStub = jest.spyOn(codeEditorRepositoryService, 'getStatus');
-                getRepositoryContentStub = jest.spyOn(codeEditorRepositoryFileService, 'getRepositoryContent');
-                subscribeForLatestResultOfParticipationStub = jest
-                    .spyOn(participationWebsocketService, 'subscribeForLatestResultOfParticipation')
-                    .mockReturnValue(subscribeForLatestResultOfParticipationSubject);
-                getFeedbackDetailsForResultStub = jest.spyOn(resultService, 'getFeedbackDetailsForResult');
-                getBuildLogsStub = jest.spyOn(buildLogService, 'getBuildLogs');
-                getFileStub = jest.spyOn(codeEditorRepositoryFileService, 'getFile');
-                saveFilesStub = jest.spyOn(codeEditorRepositoryFileService, 'updateFiles');
-                commitStub = jest.spyOn(codeEditorRepositoryService, 'commit');
-                getStudentParticipationWithLatestResultStub = jest.spyOn(programmingExerciseParticipationService, 'getStudentParticipationWithLatestResult');
-                getLatestPendingSubmissionStub = jest.spyOn(submissionService, 'getLatestPendingSubmissionByParticipationId').mockReturnValue(getLatestPendingSubmissionSubject);
-                // Mock the ResizeObserver, which is not available in the test environment
-                global.ResizeObserver = jest.fn().mockImplementation((callback: ResizeObserverCallback) => {
-                    return new MockResizeObserver(callback);
-                });
-            });
+            .overrideComponent(CodeEditorMonacoComponent, { set: { imports: [MonacoEditorComponent, MockComponent(CodeEditorHeaderComponent)] } })
```

Comment:
Only this line is new. The rest of the diff looks like that because I changed from `compileComponents.then(() => ...` to `await compileComponents(); ...`